cli/azd/pkg/azapi/stack_deployments.go

```diff
@@ -27,16 +29,35 @@ import (
 var FeatureDeploymentStacks = alpha.MustFeatureKey("deployment.stacks")
 
 const (
-	stacksPortalUrlFragment = "#@microsoft.onmicrosoft.com/resource"
+	deploymentStacksConfigKey = "DeploymentStacks"
+	stacksPortalUrlFragment   = "#@microsoft.onmicrosoft.com/resource"
 )
 
+var defaultDeploymentStackOptions = &deploymentStackOptions{
+	BypassStackOutOfSyncError: to.Ptr(false),
+	ActionOnUnmanage: &armdeploymentstacks.ActionOnUnmanage{
+		ManagementGroups: to.Ptr(armdeploymentstacks.DeploymentStacksDeleteDetachEnumDelete),
+		ResourceGroups:   to.Ptr(armdeploymentstacks.DeploymentStacksDeleteDetachEnumDelete),
+		Resources:        to.Ptr(armdeploymentstacks.DeploymentStacksDeleteDetachEnumDelete),
+	},
+	DenySettings: &armdeploymentstacks.DenySettings{
+		Mode: to.Ptr(armdeploymentstacks.DenySettingsModeNone),
+	},
+}
+
 type StackDeployments struct {
 	credentialProvider  account.SubscriptionCredentialProvider
 	armClientOptions    *arm.ClientOptions
 	standardDeployments *StandardDeployments
 	cloud               *cloud.Cloud
 }
 
+type deploymentStackOptions struct {
+	BypassStackOutOfSyncError *bool                                 `yaml:"bypassStackOutOfSyncError,omitempty"`
+	ActionOnUnmanage          *armdeploymentstacks.ActionOnUnmanage `yaml:"actionOnUnmanage,omitempty"`
```

Comment:
I'm a little worried about the approach of adding all the ARM properties to `azure.yaml` and the pass-through idea here.

Looking at what we have:

```yaml
infra:
  provider: bicep
  deploymentStacks:
    bypassStackOutOfSyncError: false
    actionOnUnmanage: 
      managementGroups: delete    # delete or detach
      resourceGroups: delete      # delete or detach
      resources: delete           # delete or detach
    denySettings:
      mode: denyDelete            # denyDelete, denyWriteAndDelete, none
      applyToChildScopes: false   # true | false
      excludedActions:            # List of actions to exclude
        - action1
        - action2
      excludedPrincipals:         # List of principals to exclude
        - principal1
        - principal2
```

I would have some questions like:

- Perhaps `bypassStackOutOfSyncError` is a one-time activity (would you really want the same behavior each time?) that would be covered by a `--force` gesture to avoid an interactive prompt?

- `actionOnUnmanage.managementGroups`. I would some time to look into the docs more, but would this only apply to management group deployments which `azd` doesn't support currently?

From an overall UX standpoint, I think we're taking the stance that "if you're here, you're an advanced user and we're just going to expose all the details of ARM". Another approach is "let's make these advanced features easier to use, or easier to understand -- what are features that a user would care about, with no Bicep or ARM specifics referenced here?"

Reply:
Yes, deployment stacks are technically an Azure resource. Currently, they’re created using the Go SDK, which exposes all the necessary configuration, although these options are mostly utilized by advanced users.

I’d like to begin exposing this in the `azure.yaml`. As we move forward with the alpha phase, we may receive additional feedback from users who might want to configure some of these settings at a higher level. Let’s hold off on drawing any conclusions until we have more data.

Reply:
My ask is that we think a little more about the scenarios and motivation behind settings in #4307. I believe this would be a fair ask, given that we would also need to document them for our public docs.

I understand that we're thinking about sourcing feedback from users. That is indeed a useful tool, but we should at least be in the 80% ballpark of correctness with 20% refinement. Here we're taking the approach that this will be a black-box 3rd-party extensible integration that azd knows little about. This hasn't been the approach with Bicep. 

Here are the concrete examples listed above with more context that we could think about:

- `bypassStackOutOfSyncError` - In the [docs](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/deployment-stacks?tabs=azure-powershell#handle-the-stack-out-of-sync-error), it calls out that as: "This switch should only be used after thoroughly review the list of resources in the stack before rerunning the command. This switch should never be used by default." I wonder if this really fits the profile of a project-level property?
- `actionOnUnmanage.managementGroups` - azd doesn't support management groups provisioning currently -- perhaps it'd be wise for us to not expose this?

I think @vhvb1989 also brought up a bigger picture question of whether these configuration are environment-specific behavior -- does `prod` require the same level of protection as `dev`? If I think from that perspective, it is perhaps true that adding the settings here into `azure.yaml` would not help as much as we'd think it would.

If we examine the established pattern from `az` is that these properties are specific to a given deployment, and environments will have different deployment stacks, we should at least consider leaning this way instead of against it. My educated but unvalidated hypothesis is that users are likely choose different settings for different environments. The proposed design here locks us into project-level configuration that we'd need to think about again when environments are brought back into the picture.

Reply:
Thanks for the detailed thoughts on this! I agree that understanding the full scope is important, but for now, I think this implementation is sufficient, especially given that we're in the early stages with this alpha feature. We want to avoid over-engineering or spending too much time refining every detail upfront when we're still in the phase of gathering user feedback.

While we may not have everything figured out from the start, this approach gives us flexibility to evolve. It doesn't lock us into any specific configuration long-term, and we can still layer in environment-level config for `azd` if it becomes necessary down the line. Similarly, we retain the option to remove certain properties later if we find they're not useful or misaligned with the project-level scope.

At this point, it's about moving forward efficiently while keeping our options open for future adjustments based on what we learn from usage and feedback.